### PR TITLE
cmd: lxc-usernsexec fixes + macro: move declarations + config_utils: macvlan fixes

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

File: src/lxc/cmd/lxc_usernsexec.c [hunk 1]
```diff
@@ -228,8 +229,20 @@ static int read_default_map(char *fnam, int which, char *user)
 			return -1;
 		}
 
-		newmap->hostid = atol(p1 + 1);
-		newmap->range = atol(p2 + 1);
+		ret = lxc_safe_ulong(p1 + 1, &newmap->hostid);
+		if (ret < 0) {
+			fclose(fin);
+			free(line);
```
Comment (Blub) [inline src/lxc/cmd/lxc_usernsexec.c]: Like the `!tmp` case below this should also `free(newmap)` - perhaps we should have an `int result` on top and just `break` here (and set it to `0` before the break at the loop's bottom) - or the usual goto...

File: src/lxc/cmd/lxc_usernsexec.c [hunk 2]
```diff
@@ -223,46 +223,37 @@ static int read_default_map(char *fnam, int which, char *user)
...
 
 		tmp = malloc(sizeof(*tmp));
-		if (!tmp) {
-			fclose(fin);
-			free(line);
-			free(newmap);
-			return -1;
-		}
+		if (!tmp)
+			goto on_error;
```
Comment (Blub) [inline src/lxc/cmd/lxc_usernsexec.c]: This now needs to set `ret = -1;` as it's 0 from the `lxc_safe_ulong()` call succeeding above.

Comment (brauner) [reply src/lxc/cmd/lxc_usernsexec.c]: ffs

Comment (brauner) [reply src/lxc/cmd/lxc_usernsexec.c]: fixed

**Pull request merged**